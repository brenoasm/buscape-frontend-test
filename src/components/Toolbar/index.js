import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import posed from "react-pose";

import { colors } from "../../theme";

import Cart from "../Cart";
import ShoppingResume from "../ShoppingResume";

const propTypes = {};

const defaultProps = {};

const StyledToolbar = styled.div`
  min-height: 80px;
  background-color: ${colors.toolbarSecondary};
  box-shadow: 0px 1px 1px ${colors.toolbarPrimary};

  > div:first-child {
    display: flex;
    align-items: center;
    padding: 15px 30px;

    > img {
      height: 72px;
    }

    > span {
      margin-left: auto;

      > span {
        padding: 1px;
        border-radius: 50%;
        background-color: ${colors.red};
        position: absolute;
        right: 60px;
        top: 50px;
        height: 28px;
        width: 28px;
        line-height: 28px;
        text-align: center;
        font-size: 20px;
        font-weight: 700;
        color: ${colors.white};
      }
    }
  }

  > div:last-child {
    background-color: ${colors.toolbarSecondary};
    overflow: hidden;
    z-index: 1000;
  }
`;

// Arrumar uma forma melhor de separar esse componente de animação
const WithSlideUpDown = posed.div({
  closed: { height: 0 },
  open: { height: "auto" }
});

const Toolbar = ({ toggleCartView, open }) => {
  return (
    <StyledToolbar>
      <div>
        <img
          src="https://imagebuscape-a.akamaihd.net/material/buscape.png"
          alt="Buscape Logo"
        />
        <span onClick={() => toggleCartView()}>
          <i className="fas fa-bars fa-3x" />
          <span>2</span>
        </span>
      </div>
      <WithSlideUpDown pose={open ? "open" : "closed"}>
        <Cart />
        <ShoppingResume />
      </WithSlideUpDown>
    </StyledToolbar>
  );
};

Toolbar.propTypes = propTypes;
Toolbar.defaultProps = defaultProps;

export default Toolbar;
