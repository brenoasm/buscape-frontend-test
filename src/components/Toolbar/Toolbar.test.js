import React from "react";
import { shallow } from "enzyme";

import Toolbar from "./";

describe("Toolbar", () => {
  it("matches the snapshot", () => {
    const wrapper = shallow(<Toolbar menuOpen={false} totalItems={0} />);

    expect(wrapper).toMatchSnapshot();
  });

  it(`shows "Não há produtos no carrinho" when totalItems is zero`, () => {
    const wrapper = shallow(<Toolbar menuOpen={false} totalItems={0} />);
    expect(wrapper.find('ForwardRef').find('div').text()).toBe("Não há produtos no carrinho");
  });

  it(`has CartContainer when totalItems is more than zero`, () => {
    const wrapper = shallow(<Toolbar menuOpen={false} totalItems={1} />);
    expect(wrapper.find('Connect(CartContainer)')).toHaveLength(1);
  });

  it(`should not show the menu when "menuOpen" is false`, () => {
    const wrapper = shallow(<Toolbar menuOpen={false} totalItems={0} />);
    expect(wrapper.find("ForwardRef").prop("pose")).toBe("closed");
  });

  it(`should show the menu when "menuOpen" is true`, () => {
    const wrapper = shallow(<Toolbar menuOpen={true} totalItems={0} />);
    expect(wrapper.find("ForwardRef").prop("pose")).toBe("open");
  });
});
