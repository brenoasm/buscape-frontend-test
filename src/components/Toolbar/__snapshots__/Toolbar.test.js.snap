// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Toolbar
    menuOpen={false}
    toggleMenuVisibility={[Function]}
    totalItems={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div>
          <Image
            alt="Buscape Logo"
            height={72}
            onBrokenImageUrl={[Function]}
            src="https://imagebuscape-a.akamaihd.net/material/buscape.png"
            width={110}
          />
          <span
            onClick={[Function]}
          >
            <i
              className="fas fa-bars fa-3x"
            />
          </span>
        </div>,
        <ForwardRef
          pose="closed"
        >
          <div>
            Não há produtos no carrinho
          </div>
        </ForwardRef>,
      ],
      "totalItems": 0,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Image
              alt="Buscape Logo"
              height={72}
              onBrokenImageUrl={[Function]}
              src="https://imagebuscape-a.akamaihd.net/material/buscape.png"
              width={110}
            />,
            <span
              onClick={[Function]}
            >
              <i
                className="fas fa-bars fa-3x"
              />
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "alt": "Buscape Logo",
              "height": 72,
              "onBrokenImageUrl": [Function],
              "src": "https://imagebuscape-a.akamaihd.net/material/buscape.png",
              "width": 110,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="fas fa-bars fa-3x"
                />,
                false,
              ],
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-bars fa-3x",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              false,
            ],
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <div>
            Não há produtos no carrinho
          </div>,
          "pose": "closed",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Não há produtos no carrinho",
          },
          "ref": null,
          "rendered": "Não há produtos no carrinho",
          "type": "div",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-EHOje",
        "isStatic": false,
        "rules": Array [
          "
  min-height: 80px;
  background-color: ",
          "#F1C71B",
          ";
  box-shadow: 0px 1px 1px ",
          "#FFCD00",
          ";
  position: absolute;
  width: 100%;
  z-index: 99999;

  > div:first-child {
    display: flex;
    align-items: center;
    padding: 15px 30px;

    > span {
      cursor: pointer;
      margin-left: auto;

      > span {
        padding: 1px;
        border-radius: 50%;
        background-color: ",
          "#E80000",
          ";
        position: absolute;
        right: 60px;
        top: 50px;
        height: 28px;
        width: 28px;
        line-height: 28px;
        text-align: center;
        font-size: 20px;
        font-weight: 700;
        color: ",
          "#FFF",
          ";
      }
    }
  }

  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-EHOje",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div>
            <Image
              alt="Buscape Logo"
              height={72}
              onBrokenImageUrl={[Function]}
              src="https://imagebuscape-a.akamaihd.net/material/buscape.png"
              width={110}
            />
            <span
              onClick={[Function]}
            >
              <i
                className="fas fa-bars fa-3x"
              />
            </span>
          </div>,
          <ForwardRef
            pose="closed"
          >
            <div>
              Não há produtos no carrinho
            </div>
          </ForwardRef>,
        ],
        "totalItems": 0,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Image
                alt="Buscape Logo"
                height={72}
                onBrokenImageUrl={[Function]}
                src="https://imagebuscape-a.akamaihd.net/material/buscape.png"
                width={110}
              />,
              <span
                onClick={[Function]}
              >
                <i
                  className="fas fa-bars fa-3x"
                />
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "alt": "Buscape Logo",
                "height": 72,
                "onBrokenImageUrl": [Function],
                "src": "https://imagebuscape-a.akamaihd.net/material/buscape.png",
                "width": 110,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="fas fa-bars fa-3x"
                  />,
                  false,
                ],
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-bars fa-3x",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                false,
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <div>
              Não há produtos no carrinho
            </div>,
            "pose": "closed",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Não há produtos no carrinho",
            },
            "ref": null,
            "rendered": "Não há produtos no carrinho",
            "type": "div",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-EHOje",
          "isStatic": false,
          "rules": Array [
            "
  min-height: 80px;
  background-color: ",
            "#F1C71B",
            ";
  box-shadow: 0px 1px 1px ",
            "#FFCD00",
            ";
  position: absolute;
  width: 100%;
  z-index: 99999;

  > div:first-child {
    display: flex;
    align-items: center;
    padding: 15px 30px;

    > span {
      cursor: pointer;
      margin-left: auto;

      > span {
        padding: 1px;
        border-radius: 50%;
        background-color: ",
            "#E80000",
            ";
        position: absolute;
        right: 60px;
        top: 50px;
        height: 28px;
        width: 28px;
        line-height: 28px;
        text-align: center;
        font-size: 20px;
        font-weight: 700;
        color: ",
            "#FFF",
            ";
      }
    }
  }

  ",
            [Function],
            "
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-EHOje",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
